### PR TITLE
ipset: fix comment and test

### PR DESCRIPTION
**Pull request by thegoodduke**
1.Add comment for every entry. The followings is sls config:

```
setname:
  ipset.set_present:
    - name: tt
    - set_type: hash:net
    - hashsize: 2048
    - comment: True
    - timeout: 0

entries:
  ipset.present:
    - set_name: tt
    - timeout: 0
    - entry:
        - 192.168.1.0/24 comment "hello"
        - 192.168.2.0/32 comment "hello"
```

2.fix test=True.
To avoid some diff is not reported.
For example, the machine has entries: ["192.168.1.1"]. the sls config's entry: ["192.168.1.1, 192.168.1.2"]

### CODE REVIEW

File: salt/states/ipset.py [hunk 1]
```diff
@@ -198,19 +199,25 @@ def present(name, entry=None, family='ipv4', **kwargs):
...
+            if 'comment' in _entry:
+                _entry = '{0} timeout {1} {2} {3}'.format(entry.split()[0], kwargs['timeout'], entry.split()[1], entry.split()[2])
+            else:
+                _entry = '{0} timeout {1}'.format(entry.split()[0], kwargs['timeout'])
 
         if __salt__['ipset.check'](kwargs['set_name'],
                                    _entry,
                                    family) is True:
-            ret['result'] = True
+            if test_flag == False:
```
Comment (basepi) [inline salt/states/ipset.py]: Please use `if test_flag is False:` as it's a little bit faster. For True/False/None, you can use `is` for comparison.

**Comment by basepi** [comment]
Your changes to salt/modules/ipset.py changed that whole block to a 5 space indent. Salt uses 4 space indents. I also left a comment inline.

File: salt/states/ipset.py (hunk 1 above)
Comment (thegoodduke) [inline salt/states/ipset.py]: Thanks! I have fixed it.

File: salt/modules/ipset.py [hunk 2]
```diff
@@ -430,11 +430,13 @@ def check(set=None, entry=None, family='ipv4'):
...
-        if entry.find('-') != -1 and entry.count('-') == 1:
-            start, end = entry.split('-')
+        _entry = entry.split()[0]
+        _entry_extra = entry.split()[1:]
+        if _entry.find('-') != -1 and _entry.count('-') == 1:
+            start, end = _entry.split('-')
 
             if settype == 'hash:ip':
-                entries = [str(ipaddress.ip_address(ip)) for ip in long_range(
+                entries = [' '.join([str(ipaddress.ip_address(ip)), ' '.join(_entry_extra)]) for ip in log_range(
```
Comment (basepi) [inline salt/modules/ipset.py]: typo here. s/log_range/long_range/

Comment (thegoodduke) [inline salt/modules/ipset.py]: fix it.

**Comment by basepi** [comment]
thanks @thegoodduke.

**Comment by thegoodduke** [comment]
I think I should not push to develop branch. The bug I have fixed can be mereged to relevant version?
1.Add comment for every entry. The followings is sls config:
 Add to the tag:2015.8. the bug is imported by  0a499bd0fa4de916c181d5c894eefcc515dea75e.

2.fix test=True:
 Add to the tag:2015.5. the bug is imported by 6f3f446e5908628a289ae5416ea8d069e0177ef2.

Need I commit it, again?

Thanks.

**Comment by basepi** [comment]
Because you combined two fixes on a single commit, yes, you'll have to make new pull requests against 2015.8 and 2015.5, because we don't want to backport both fixes to 2015.5.

**Pull request merged**